src/pudl_archiver/orchestrator.py

```diff
@@ -191,28 +190,29 @@ async def run(self):
         # Leave immediately after generating changes if dry_run
         if self.dry_run:
             logger.info("Dry run, aborting")
-            return self.new_deposition
+            return Unchanged(dataset_name=self.data_source_id, reason="Dry run.")
 
         await self._apply_changes()
         self.new_deposition = await self.depositor.get_record(self.new_deposition.id_)
         new_datapackage, old_datapackage = await self._update_datapackage(resources)
         await self._apply_changes()
 
         if self.changed:
-            validation_success = self.downloader.validate_archive(
```

Comment:
The first structure that jumps to mind is

```
orchestrator
    |--> downloader
    |--> validator/summarizer
    |--> uploader
```

Where the three "thing-doers" are siblings and the orchestrator tells them what to do.

What were you considering when thinking about where to put the validation/summary stuff?

Reply:
The reason I put the validator/summarizer stuff in the downloader is so each dataset can override the `dataset_validate_archive` method to add one off validation tests. One other pattern I could see is adding a decorator for adding validation tests, so tests could still be defined in the same module as a datasets archiver without needing them to be in the class itself. Something like:

```py
class Eia860Archiver(AbstractDatasetArchiver):
    # Do downloading stuff


@validation_test(dataset_name="eia860")
def test_something():
    return True
```

Do you think this pattern would make more sense? It would allow us to keep the downloading/validating/uploading separate

Reply:
Ah, so basically if we have the downloader & the validator/summarizer be different classes, we end up with two dataset-specific classes per dataset. Which is annoying for sure. I think I'd rather keep everything in the same class than have a bunch of dangling validation tests in the module, that separation doesn't seem useful enough to justify the sort of strange pattern.

Reply:
Yeah ideally we'll get to the place where if you subclass the `AbstractDatasetArchiver` and implement a couple of basic methods, you'll end up with new archives being auto-generated  and not need to think about anything else